tensorflow_datasets/core/features/feature.py

```diff
@@ -205,6 +213,44 @@ def dtype(self):
     """Return the dtype (or dict of dtype) of this FeatureConnector."""
     return tf.nest.map_structure(lambda t: t.dtype, self.get_tensor_info())
 
+  @classmethod
+  @abc.abstractmethod
+  def from_json(cls, value) -> 'FeatureConnector':
+    """Return the FeatureConnector by decoding the JSON string."""
+    feature_dict = {}
+    for feature_name, feature in value['content']:
+      subclass = cls._REGISTERED_FEATURES.get(feature_name)
+      if subclass is None:
+        raise ValueError(
+            f'Invalid feature name {feature_name} with feature: {feature}\n'
+            f'Supported: {list(cls._REGISTERED_FEATURES)}'
+        )
+      feature_dict.update(subclass.from_json(feature))
+    return feature_dict
```

Comment:
Just for info, you're returning a `dict`, but the return type should be `FeatureConnector`